BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsEvent.kt

```diff
@@ -1,44 +1,13 @@
 package com.braintreepayments.api
 
-import androidx.room.PrimaryKey
-import androidx.room.ColumnInfo
-import androidx.room.Entity
-
-// NEXT MAJOR VERSION: Convert to data class, we're unable to do so now because the
-// counterpart Java class is technically extendable, and making this a data class would
-// result in a breaking change
-
-// NEXT MAJOR VERSION: remove open modifiers
-
-@Entity(tableName = "analytics_event")
-open class AnalyticsEvent internal constructor(
-    open val name: String,
-
-    @ColumnInfo(name = "paypal_context_id")
-    open val payPalContextId: String? = null,
-
-    @ColumnInfo(name = "link_type")
-    open val linkType: String? = null,
-
-    @ColumnInfo(name = "venmo_installed", defaultValue = "0")
-    open val venmoInstalled: Boolean = false,
-
-    @ColumnInfo(name = "is_vault", defaultValue = "0")
-    open val isVaultRequest: Boolean = false,
-
-    @ColumnInfo(name = "start_time", defaultValue = "-1")
-    open val startTime: Long? = -1,
-
-    @ColumnInfo(name = "end_time", defaultValue = "-1")
-    open val endTime: Long? = -1,
-
-    @ColumnInfo(name = "endpoint")
-    open val endpoint: String? = null,
-
-    open val timestamp: Long = System.currentTimeMillis()
-) {
-    @JvmField
-    @PrimaryKey(autoGenerate = true)
-    @ColumnInfo(name = "_id")
-    var id: Long = 0L
-}
+internal data class AnalyticsEvent constructor(
```

Comment:
_technically_ a breaking change, but since no merchant should be using this class I am ok with it - what do others think?

Reply:
Yeah it felt like the right call to break it. I'll add a `@RestrictTo` annotation to call out that this is intended for SDK use only. I doubt anyone's using / extending these types. If they are I think it's better for us to break those integrations to discourage its use.

Reply:
i'm also good with this breaking change. we shouldn't need the restrict to annotation since the entire class is marked as internal.

Reply:
Oh true yeah. I'll remove.